sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/ResolveDefaultColumns.scala

```diff
@@ -15,139 +15,385 @@
  * limitations under the License.
  */
 
-package org.apache.spark.sql.catalyst.util
-
 import org.apache.spark.sql.AnalysisException
+import org.apache.spark.sql.catalyst.TableIdentifier
 import org.apache.spark.sql.catalyst.analysis.Analyzer
+import org.apache.spark.sql.catalyst.catalog.{SessionCatalog, UnresolvedCatalogRelation}
 import org.apache.spark.sql.catalyst.expressions._
 import org.apache.spark.sql.catalyst.optimizer.ConstantFolding
 import org.apache.spark.sql.catalyst.parser.{CatalystSqlParser, ParseException}
 import org.apache.spark.sql.catalyst.plans.logical._
+import org.apache.spark.sql.catalyst.rules.Rule
+import org.apache.spark.sql.catalyst.util.ResolveDefaultColumns._
 import org.apache.spark.sql.internal.SQLConf
 import org.apache.spark.sql.types._
 
-/**
- * This object contains fields to help process DEFAULT columns.
- */
-object ResolveDefaultColumns {
-  // This column metadata indicates the default value associated with a particular table column that
-  // is in effect at any given time. Its value begins at the time of the initial CREATE/REPLACE
-  // TABLE statement with DEFAULT column definition(s), if any. It then changes whenever an ALTER
-  // TABLE statement SETs the DEFAULT. The intent is for this "current default" to be used by
-  // UPDATE, INSERT and MERGE, which evaluate each default expression for each row.
-  val CURRENT_DEFAULT_COLUMN_METADATA_KEY = "CURRENT_DEFAULT"
-  // This column metadata represents the default value for all existing rows in a table after a
-  // column has been added. This value is determined at time of CREATE TABLE, REPLACE TABLE, or
-  // ALTER TABLE ADD COLUMN, and never changes thereafter. The intent is for this "exist default"
-  // to be used by any scan when the columns in the source row are missing data. For example,
-  // consider the following sequence:
-  // CREATE TABLE t (c1 INT)
-  // INSERT INTO t VALUES (42)
-  // ALTER TABLE t ADD COLUMNS (c2 INT DEFAULT 43)
-  // SELECT c1, c2 FROM t
-  // In this case, the final query is expected to return 42, 43. The ALTER TABLE ADD COLUMNS command
-  // executed after there was already data in the table, so in order to enforce this invariant,
-  // we need either (1) an expensive backfill of value 43 at column c2 into all previous rows, or
-  // (2) indicate to each data source that selected columns missing data are to generate the
-  // corresponding DEFAULT value instead. We choose option (2) for efficiency, and represent this
-  // value as the text representation of a folded constant in the "EXISTS_DEFAULT" column metadata.
-  val EXISTS_DEFAULT_COLUMN_METADATA_KEY = "EXISTS_DEFAULT"
-
+package org.apache.spark.sql.catalyst.analysis {
```

Comment:
Normally we don't do this.
Let's move `case class ResolveDefaultColumns` to a new file under `org.apache.spark.sql.catalyst.analysis`.
We can keep `object ResolveDefaultColumns` here. My suggestion is to rename it as `ResolveDefaultColumnUtil`

Reply:
SG, done.